ntrans_dataprep.py

```diff
@@ -17,36 +18,37 @@
 """
 
 
-def write_data_to_csv(
-    counted_two_grams,
-    counted_three_grams,
-    counted_four_grams,
-    counted_five_grams,
-    counted_six_grams,
-):
+def write_data_to_csv(n_to_ngrams):
     """
     Writes the N-grams to CSV files. These CSV files are read from ntrans.py
     """
-    pass
 
+    filenames = ["2-grams", "3-grams", "4-grams", "5-grams", "6-grams"]
+    pathlib.Path("ngrams").mkdir(exist_ok=True)
 
-def count_ngram_frequency(twograms, threegrams, fourgrams, fivegrams, sixgrams):
+    for n, counter in n_to_ngrams.items():
+
+        # "n - 2" to get index of "filenames"
+        file_path = "./ngrams/" + filenames[n - 2] + ".csv"
+
+        with open(file_path, mode="w") as write_data_file:
+            data_writer = csv.writer(write_data_file, delimiter=",")
+
+            for ngram, count in counter.most_common():
+                csv_row = [" ".join(ngram), str(count)]
+                data_writer.writerow(csv_row)
+
+
+def count_ngram_frequency(n_to_ngrams):
     """
     Counts the frequency of each N-gram to distinguish the most common ones.
     """
-    counted_two_grams = collections.Counter(twograms)
-    counted_three_grams = collections.Counter(threegrams)
-    counted_four_grams = collections.Counter(fourgrams)
-    counted_five_grams = collections.Counter(fivegrams)
-    counted_six_grams = collections.Counter(sixgrams)
-
-    write_data_to_csv(
-        counted_two_grams,
-        counted_three_grams,
-        counted_four_grams,
-        counted_five_grams,
-        counted_six_grams,
-    )
+
+    for n in n_to_ngrams.keys():
+        n_to_ngrams[n] = collections.Counter(n_to_ngrams[n])
+
+    # print(n_to_ngrams[2])
+    write_data_to_csv(n_to_ngrams)
```

Comment:
This is what I meant yesterday:

```suggestion
    write_data_to_csv({
        n: collections.Counter(ngrams)
        for n, ngrams in n_to_ngrams.items()
    })
```